Userland/Libraries/LibWeb/Layout/FlexFormattingContext.cpp

```diff
@@ -19,49 +24,666 @@ FlexFormattingContext::~FlexFormattingContext()
+    };
+    auto is_cross_auto = [&is_row](Box& box) {
+        return is_row ? box.computed_values().height().is_auto() : box.computed_values().width().is_auto();
+    };
+    auto is_main_axis_margin_first_auto = [&is_row](Box& box) {
+        return is_row ? box.computed_values().margin().left.is_auto() : box.computed_values().margin().top.is_auto();
+    };
+    auto is_main_axis_margin_second_auto = [&is_row](Box& box) {
+        return is_row ? box.computed_values().margin().right.is_auto() : box.computed_values().margin().bottom.is_auto();
+    };
+    auto sum_of_margin_padding_border_in_main_axis = [&is_row](Box& box) {
+        if (is_row) {
+            return box.box_model().margin.left
+                + box.box_model().margin.right
+                + box.box_model().padding.left
+                + box.box_model().padding.right
+                + box.box_model().border.left
+                + box.box_model().border.right;
+        } else {
+            return box.box_model().margin.top
+                + box.box_model().margin.bottom
+                + box.box_model().padding.top
+                + box.box_model().padding.bottom
+                + box.box_model().border.top
+                + box.box_model().border.bottom;
+        }
+    };
+    auto calculate_hypothetical_cross_size = [&is_row, this](Box& box) {
+        // FIXME: Don't use BFC exclusively, there are more FormattingContexts.
+        if (is_row) {
+            return BlockFormattingContext::compute_theoretical_height(box);
+        } else {
+            // FIXME: This is very bad.
+            BlockFormattingContext context(box, this);
+            context.compute_width(box);
+            return box.width();
+        }
+    };
+    auto set_main_size = [&is_row](Box& box, float size) {
+        if (is_row)
+            box.set_width(size);
+        else
+            box.set_height(size);
+    };
+    auto set_cross_size = [&is_row](Box& box, float size) {
+        if (is_row)
+            box.set_height(size);
+        else
+            box.set_width(size);
+    };
+    auto set_offset = [&is_row](Box& box, float main_offset, float cross_offset) {
+        if (is_row)
+            box.set_offset(main_offset, cross_offset);
+        else
+            box.set_offset(cross_offset, main_offset);
+    };
+    auto set_main_axis_first_margin = [&is_row](Box& box, float margin) {
+        if (is_row)
+            box.box_model().margin.left = margin;
+        else
+            box.box_model().margin.top = margin;
+    };
+    auto set_main_axis_second_margin = [&is_row](Box& box, float margin) {
+        if (is_row)
+            box.box_model().margin.right = margin;
+        else
+            box.box_model().margin.bottom = margin;
+    };
 
-    // Then, place the items on a vertical or horizontal line.
-    float x = 0;
-    float y = 0;
-    float tallest_child_height = 0;
-    float widest_child_width = 0;
+    /* 1. Generate anonymous flex items */
```

Comment:
More of these C-style comments here.

Reply:
Shit, force-pushing bit me here. I fixed Klings comments at work (my machine there is just faster...) and then continued at home, I didn't pull when I got home.